src/libraries/System.Net.Http/src/System/Net/Http/HttpClient.cs

```diff
@@ -726,14 +736,6 @@ private void CheckDisposed()
             }
         }
 
-        private static void CheckRequestMessage(HttpRequestMessage request)
```

Comment:
This has been discussed between @MihaZupan and @stephentoub in our chat.

Originally, I started removing [`_sendStatus`](https://github.com/dotnet/runtime/pull/41022/files#diff-4ec2e0e7f4a0ab4e4aaf56310b6b76d8L21) from `HttpRequestMessage`, thinking it's only used for telemetry. To later discover it's also used for this check. I took the liberty of removing it since that was Stephen's preference in the chat and I've already had it mostly removed. 

It's NOT necessary for this PR to work and I can revert it.